src/fix_hp.cpp

```diff
@@ -0,0 +1,798 @@
+/* ----------------------------------------------------------------------
```

Comment:
*NEVER* commit code that belongs to an "installed package". in fact, instead the names of the files in the package should be added to the `.gitignore` file in the `src` folder, so that this won't happen by accident.